### ISSUE TITLE
Gas readings are much higher than they should be

### ISSUE THREAD

**Issue by robhawkes** [megakid/ha_hildebrand_glow_ihd_mqtt#6]
I've recently set up my new Glow CAD with Home Assistant using this setup and on the whole it's been great. Thanks!

One thing I'm trying to sort out is my gas readings being completely off in Home Assistant vs. what is shown in the Bright app, on the Glow CAD and on my IHD.

Here's an example of my gas usage today as it looks in Home Assistant:
![image](https://user-images.githubusercontent.com/22612/174404658-fea88e10-fe12-4b50-9283-8f1d04ec5d33.png)
 
And today in Bright:
![image](https://user-images.githubusercontent.com/22612/174404722-a3f690cc-6a74-4715-843e-1bac47818134.png)

As you can see the usage reported in Home Assistant is reporting hourly values in the many hundreds of millions of kWh vs. values of <5kWh for the same period in Bright, Glow CAD and the IHD.

Here's how things look in Home Assistant from the sensor data:
![image](https://user-images.githubusercontent.com/22612/174405783-adceadc1-374d-4597-822d-f666e33d054d.png)

It looks like month, week and today values for gas are correct and line up with the other devices, it's just the total gas import that's way off. The meter is also quite new and the other devices report a _lifetime_ gas usage of <1,000 kWh on that meter so far.

Any suggestions on how I can further diagnose this and get the correct readings so I can have the right values showing on the energy dashboard in Home Assistant?

**Comment by megakid** [comment]
Can you connect MQTT Explorer to your MQTT and paste in the raw messages
under your gasmeter topic?

http://mqtt-explorer.com

In theory the monthly usage is suitable for the energy dashboard but might
need a different categorisation (total_increasing) - post your messages and
we’ll go from there

**Comment by robhawkes** [comment]
Sure thing, here's the raw payload from MQTT Explorer. Looks like it's reporting the incorrect cumulative import for whatever reason, though the day, week and month values are correct.

```
{
  "gasmeter": {
    "timestamp": "2022-06-18T08:39:21Z",
    "energy": {
      "export": {
        "cumulative": 0,
        "units": "kWh"
      },
      "import": {
        "cumulative": 711118290944,
        "day": 3.045,
        "week": 34.202,
        "month": 178.53,
        "units": "kWh",
        "mprn": "REMOVED",
        "supplier": "---",
        "price": {
          "unitrate": 0.0736,
          "standingcharge": 0.2722
        }
      }
    },
    "power": {
      "value": 0,
      "units": "kW"
    }
  }
}
```

I did try using the monthly gas import values on the Energy dashboard but it complained about there not being a `last_reset` value so I think that lines up with what you said about `total_increasing` based on my limited knowledge of the charts.

If we can get the month/week/day values into the Energy dashboard as `total_increasing` (which I believe auto-detects periodic resets) then I can try following up about the high cumulative gas value with Hildebrand directly, unless you have ideas about that bit.

```
Last reset missing
The following entities have state class 'measurement' but 'last_reset' is missing:
sensor.smart_meter_gas_import_this_month
```

**Comment by robhawkes** [comment]
[I've posted in the Glow forum](https://forum.glowmarkt.com/index.php?p=/discussion/192/mqtt-cad-cumulative-gas-readings-are-off-by-hundreds-of-millions-of-kwh) about the incorrect gas meter read as I found out it's also showing as incorrect on the Glow Display & CAD. The meter read (in m3) is fine on my supplier's IHD.

**Comment by megakid** [comment]
Thanks for flagging the total_increasing thing.  I've just released v1.4.0 which fixes this. [49c3171](https://github.com/megakid/ha_hildebrand_glow_ihd_mqtt/commit/49c31712b96e24cad5bc1b9a7cddd50ccd28381a)

**Comment by robhawkes** [comment]
Amazing! Thanks for sorting that out. I've updated my Energy dashboard to use `smart_meter_gas_import_this_month` and it didn't complain this time (though waiting for HA to populate the dashboard again), so I'm hoping that'll allow me to work around the bad cumulative meter read data).
